build-system/tasks/check-types.js

```diff
@@ -122,7 +167,10 @@ const TYPE_CHECK_TARGETS = {
   // introduced. It is okay to remove a file from this list only when fixing a
   // bug for cherry-pick.
   'pride': {
-    srcGlobs: PRIDE_FILES_GLOBS,
+    srcGlobs: [
+      ...PRIDE_FILES_GLOBS,
+      ...globby.sync('src/core/**/*.js').filter((p) => !p.includes('extern')),
```

Comment:
Two bits: `TYPE_CHECK_TARGETS` should be an object literal, this will make it scan the file system on import. When dynamic values are needed, let's make it an anon function as the `extensions` target is.

Second, let's leave this as `srcGlobs: PRIDE_FILES_GLOBS` and just add `'src/core/**/*.js` to `PRIDE_FILES_GLOBS`. Anything in `externGlobs` is automatically excluded from sources, so this additional logic is unnecessary.

Reply:
Thats what I thought too, but it doesn't work / throws. Example incoming

Reply:
![Screen Shot 2021-04-29 at 11 39 49 AM](https://user-images.githubusercontent.com/4656974/116578664-a2f3eb80-a8df-11eb-9300-b5f90deb1e96.png)

Reply:
I'm not sure the glob exclusion technique worked. I've replaced with `.filter()`: https://github.com/ampproject/amphtml/pull/33957/commits/060b66a47443b524fbd2822ae446fb4fea5bdcbb

Reply:
I'm not clear on what the exact code was that generated this error, but using exclusion globs WAI:
![image](https://user-images.githubusercontent.com/6694512/116583828-bfdeed80-a8e4-11eb-854e-34a696369985.png)
Could you share the snippet that created these errors?

Reply:
The `extraGlobs` containing the exclusions were not being passed to globby, but were being directly passed to Closure

Reply:
I believe that's incorrect. Currently your changes to that logic are a no-op AFAICT:
![image](https://user-images.githubusercontent.com/6694512/116588497-a4c2ac80-a8e9-11eb-9c9c-0ce49f1d7009.png)

As for the `extraGlobs`, as long as `opts.noAddDeps` is set they are directly consumed by `globby.sync` when producing the set of `srcs` that then get passed to Closure
https://github.com/ampproject/amphtml/blob/main/build-system/compile/compile.js#L416-L422

Reply:
With the exception of the extra call to globby.sync, which doesn't need to be there

Reply:
TIL the order matters. As shown, it worked. But if you put the externs first, it breaks.

![Screen Shot 2021-04-29 at 1 22 37 PM](https://user-images.githubusercontent.com/4656974/116592464-084ed900-a8ee-11eb-8f44-49cf5e20aa00.png)